snmp/datadog_checks/snmp/snmp.py

```diff
@@ -500,6 +502,82 @@ def report_metrics(
                 metric_tags = tags + metric.tags
                 self.submit_metric(name, val, metric.forced_type, metric_tags, metric.options)
 
+    def report_bandwidth_use_metric(
```

Comment:
Seems there are a lot of complexity/logic for an add-on feature. It's not easy to have good coverage for all the cases to be highly confident there is no issue.

It seems there is possibly ways to reduce the complexity, but I don't want to block things because of that.

Reply:
Is it still the case?